cpp/src/io/fst/device_dfa.cuh

```diff
@@ -70,6 +68,7 @@ cudaError_t DeviceTransduce(void* d_temp_storage,
                             TransducedIndexOutItT transduced_out_idx_it,
                             TransducedCountOutItT d_num_transduced_out_it,
                             uint32_t seed_state = 0,
+                            uint32_t* final_state = nullptr,
```

Comment:
that makes a lot of sense. Since the previous interface requires the initial state to be host-accessible, maybe we need to find a balance between providing multiple overloads and having a clean interface.